src/internal_rep.rs

```diff
@@ -1549,6 +1687,73 @@ impl<'a> FunctionInfo<'a> {
         })
     }
 
+    // Create a derived FunctionInfo from the union of derive_classes
+    // derive_classes may be a set of classes to be unioned, or a single class to use
+    // All provided classes are to be unioned (the decision to union all vs use a single one was
+    // made in the parent)
+    pub fn new_derived_function(
+        name: &CascadeString,
+        deriving_type: &'a TypeInfo,
+        derive_classes: &Vec<&CascadeString>,
+        functions: &FunctionMap<'a>,
+        file: &'a SimpleFile<String, String>,
+    ) -> Result<FunctionInfo<'a>, CascadeErrors> {
+        let mut first_parent = None;
+        let mut derived_body = BTreeSet::new();
+        for parent in derive_classes {
+            // The parent may or may not have such a function implemented.
+            // As long as at least one parent has it, that's fine
+            let parent_function = match functions.get(&get_cil_name(Some(parent), name)) {
+                Some(f) => f,
+                None => continue,
+            };
+
+            // All parent functions must have the same prototype.  If this is the first function we
+            // are looking at, we save that prototype.  Otherwise, we compare to ensure they are
+            // identical
+            match first_parent {
+                None => first_parent = Some(parent_function),
+                Some(first_parent) => {
+                    if parent_function.args != first_parent.args {
```

Comment:
I guess in a future evolution, Cascade could allow same argument types but with different names.

Reply:
I believe that is the case in this PR.  See `impl PartialEq for FunctionArgument`.

Reply:
Indeed. What append if arguments have the same type but not the same name? I don't see a "name conversion" nor related tests.

Reply:
Yeah, good catch.  I suspect we're broken in that situation.  I'll update for the next push.

I expect the exact name doesn't matter in that situation, just that it needs to be consistent.  So we should be able to simply choose a name from one of the functions (or come up with our own name that we know is invalid elsewhere).  Since the name used is internal to this function and only appears in compiled output.

I guess the question there would be if it ever shows up in error messages or future analysis/diagnostic tools.  Maybe something consistent like "_arg1", "_arg2" would be sensible across the board in all derived functions?

Reply:
Choosing the right name generation is indeed worth thinking about because of error messages. I think we should try to keep as much as possible the original names, maybe concatenating them in a deterministic way (`sort -u`-like with `_` in between)?

Reply:
I think that seems reasonable.  I've pushed a new commit that implements that.